src/Microsoft.Performance.SDK/Processing/ITableDiscoverer.cs

```diff
@@ -0,0 +1,309 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Performance.SDK.Extensibility;
+
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    ///     Provides an interface for providing tables that are exposed
+    ///     by <see cref="ProcessingSource"/>s.
+    /// </summary>
+    public interface ITableProvider
+    {
+        /// <summary>
+        ///     Returns a collection of tables.
+        /// </summary>
+        /// <param name="tableConfigSerializer">
+        ///     The serializer used to deserialize table configurations.
+        /// </param>
+        /// <returns>
+        ///     A collection of tables.
+        /// </returns>
+        ISet<DiscoveredTable> Discover(ISerializer tableConfigSerializer);
+    }
+
+    /// <summary>
+    ///     Provides common <see cref="ITableProvider"/> implementations.
+    /// </summary>
+    public static class TableDiscovery
+    {
+        /// <summary>
+        ///     Creates a new <see cref="ITableProvider"/> instance that will
+        ///     discover tables in the given <see cref="Assembly"/>. Tables are
+        ///     discovered in the assembly by being a type decorated with the
+        ///     <see cref="TableAttribute"/> attribute.
+        /// </summary>
+        /// <param name="assembly">
+        ///     The <see cref="Assembly"/> to search for tables.
+        /// </param>
+        /// <returns>
+        ///     A new <see cref="ITableProvider"/> that discovers tables in the
+        ///     specified <paramref name="assembly"/>.
+        /// </returns>
+        public static ITableProvider CreateForAssembly(Assembly assembly)
```

Comment:
would it make sense to also have methods `CreateForAssembly(IProcessingSource)` and `CreateForNamespace(IProcessingSource)` that use the processing source's assembly/namespace? I could imagine that being a common scenario

Reply:
Yes, that is a good idea. I will add it.